internal/oci/annotations.go

```diff
@@ -55,6 +56,19 @@ func ProcessAnnotations(ctx context.Context, s *specs.Spec) (err error) {
 		}
 	}
 
+	// validate host process containers annotations are not conflicting
+	disableHPC := ParseAnnotationsBool(ctx, s.Annotations, annotations.DisableHostProcessContainer, false)
```

Comment:
This is probably the only annotation that needs to be handled like this for now. In the future we may want to consider defining a "conflict annotations map" or similar, e.g.
```
conflictingAnnotations := map[string]string{
    annotations.DisableHostProcessContainer: annotations.HostProcessContainer,
}

for k, v := range conflictingAnnotations {
    v1 := ParseAnnotationsBool(ctx, s.Annotations, k, false)
    v2 := ParseAnnotationsBool(ctx, s.Annotations, v, false)
    if v1 && v2 {
        // handle conflict
    }
}
```

Maybe it's worth just doing it now too, rather than special casing it to HPC? thoughts?